server/channels/store/sqlstore/channel_store_categories.go

```diff
@@ -741,6 +802,8 @@ func (s SqlChannelStore) UpdateSidebarCategories(userId, teamId string, categori
 			copy(destCategory.Channels, category.Channels)
 
 			destCategory.Muted = category.Muted
+		} else {
+			destCategory.Channels = make([]string, 0)
```

Comment:
This is a minor detail, but the previous version of `completePopulatingCategoryChannelsT` would always set the DMs categories to contain an empty slice of channel IDs instead of a nil one. I made that explicit to avoid having to update the tests